lib/screens/cart_page.dart

```diff
@@ -955,18 +955,30 @@ class _CartPageState extends State<CartPage> {
   }
 
   Widget _buildQuantitySelector(
-    String productDocumentId,
+    Product product,
     StockItem stockRequest,
   ) {
-    var controller = QuantityController(value: stockRequest.quantity);
+    var maxCounterValue = 10;
```

Comment:
L961-L968 

```
var maxCounterValue = 10;
    var requestedProductStockItem = product.stock.items.where((stockElement) => stockElement.quantity > 0 && stockElement.size == stockRequest.size && stockElement.color == stockRequest.color);
    if(requestedProductStockItem.length > 0){
        maxCounterValue = requestedProductStockItem.first.quantity < 10 ? requestedProductStockItem.first.quantity : 10;
    }else{
      requestedProductStockItem = product.stock.items.where((stockElement) => stockElement.quantity > 0 &&  stockElement.color == stockRequest.color);
      maxCounterValue = requestedProductStockItem.first.quantity < 10 ? requestedProductStockItem.first.quantity : 10;
    }
```

can be replaced by:

```
var requestedProductStockItem = product.stock.items.where((stockElement) => stockElement.quantity > 0 && stockElement.size == stockRequest.size && stockElement.color == stockRequest.color);
if (requestedProductStockItem.length > 0) {
  requestedProductStockItem = product.stock.items.where((stockElement) => stockElement.quantity > 0);
}
int stockQuantity = requestedProductStockItem.first.quantity;
int maxCounterValue = stockQuantity < 10 ? stockQuantity : 10;
```

Reply:
This suggested change works fine only when all items in cart page have StockType = 0. Maybe having the stocktype in stockRequest item could make this filter easier and shorter

Reply:
`stockType` is available in `product.stock`. By definition, all `stockItems` in a product have the same `stockType`!

Reply:
```
if (product.stock.stockType == SizeAndColor)
{
  requestedProductStockItem = product.stock.items.where((stockElement) => stockElement.size == stockRequest.size && stockElement.color == stockRequest.color && stockElement.quantity > 0);
}
else if (product.stock.stockType == SizeOnly)
{
  requestedProductStockItem = product.stock.items.where((stockElement) => stockElement.size == stockRequest.size && stockElement.quantity > 0);
}
else if (product.stock.stockType == ColorOnly)
{
  requestedProductStockItem = product.stock.items.where((stockElement) => stockElement.color == stockRequest.color && stockElement.quantity > 0);
}
else (product.stock.stockType == QuantityOnly)
{
  requestedProductStockItem = product.stock.items.where((stockElement) => stockElement.quantity > 0);
}
```